lib/peers.go

```diff
@@ -88,8 +98,17 @@ func (m PeerMethods) Disconnect(ctx context.Context, p *ConnectParamsPod) error
 
 // ConnectionsParams defines parameters for the Connections method
 type ConnectionsParams struct {
-	Limit  int `json:"limit"`
-	Offset int `json:"offset"`
+	params.List `json:"list,omitempty"`
+}
+
+// SetNonZeroDefaults sets a default limit and offset
+func (p *ConnectionsParams) SetNonZeroDefaults() {
+	if p.Offset < 0 {
+		p.Offset = 0
+	}
+	if p.Limit <= 0 {
+		p.Limit = params.DefaultListLimit
+	}
 }
```

Comment:
embedding `params.List` has the advantage of us being able to write unique `SetNonZeroDefaults`, in case we want different limits for different endpoints.